src/modules/scraper/adapter/messaging/FillEventsConsumer.ts

```diff
@@ -19,56 +19,91 @@ export class FillEventsConsumer {
-        if (diffMinutes >= 20) {
-          await this.scraperQueuesService.publishMessage<BlocksEventsQueueMessage>(ScraperQueue.BlocksEvents, {
-            chainId: originChainId,
-            from: olderDeposit.blockNumber,
-            to: newerDeposit.blockNumber,
-          });
-        }
-      }
+      await this.tryToRefetchDepositEvents(job.data);
       throw new Error("Deposit not found in db");
     }
 
-    const fillTxIndex = deposit.fillTxs.findIndex(
-      (fillTx) => fillTx.hash === transactionHash && fillTx.totalFilledAmount === totalFilledAmount,
-    );
-
-    if (fillTxIndex !== -1) {
+    if (this.fillTxAlreadyProcessed(deposit, job.data)) {
       this.logger.warn("Fill event already processed");
       return;
     }
 
-    deposit.realizedLpFeePct = new BigNumber(deposit.realizedLpFeePct).plus(realizedLpFeePct).toString();
-    // for referrals, the realized lp fee should be capped at 0.12%
-    const maxRealizedLpFeePct = new BigNumber(10).pow(18).times(0.0012);
-    deposit.realizedLpFeePctCapped = BigNumber.min(deposit.realizedLpFeePct, maxRealizedLpFeePct).toString();
+    deposit.fillTxs = [
+      ...deposit.fillTxs,
+      { fillAmount, hash: transactionHash, isSlowRelay, realizedLpFeePct, totalFilledAmount },
+    ];
+    const bridgeFeePct = this.computeBridgeFee(deposit, job.data);
 
     if (new BigNumber(deposit.filled).lt(totalFilledAmount)) {
       deposit.filled = totalFilledAmount;
     }
 
-    deposit.fillTxs = [...deposit.fillTxs, { hash: transactionHash, totalFilledAmount }];
     deposit.status = new BigNumber(deposit.amount).eq(deposit.filled) ? "filled" : "pending";
+    deposit.bridgeFeePct = bridgeFeePct.toString();
 
     await this.depositRepository.save(deposit);
   }
 
+  private computeBridgeFee(deposit: Deposit, fill: FillEventsQueueMessage) {
+    const maxBridgeFeePct = new BigNumber(10).pow(18).times(0.0012);
+    const validFills = deposit.fillTxs.filter((fill) => !fill.isSlowRelay); // all fills associated with a deposit that are NOT slow fills
+    const relayerFeePctChargedToUser = validFills.reduce((cumulativeFee, fill) => {
+      const relayerFee = new BigNumber(fill.fillAmount).multipliedBy(deposit.depositRelayerFeePct);
+      return relayerFee.plus(cumulativeFee);
+    }, new BigNumber(0));
+    const blendedRelayerFeePct = relayerFeePctChargedToUser.dividedBy(deposit.amount).decimalPlaces(0, 1);
+    const bridgeFeePct = blendedRelayerFeePct.plus(fill.realizedLpFeePct);
+    const bridgeFeePctCapped = BigNumber.min(bridgeFeePct, maxBridgeFeePct);
+
+    return bridgeFeePctCapped;
+  }
+
+  private fillTxAlreadyProcessed(deposit: Deposit, fill: FillEventsQueueMessage) {
+    const { totalFilledAmount, transactionHash } = fill;
+    const fillTxIndex = deposit.fillTxs.findIndex(
+      (fillTx) => fillTx.hash === transactionHash && fillTx.totalFilledAmount === totalFilledAmount,
```

Comment:
This might not be totally correct. Imagine one transaction hash has multiple fills for _different_ deposits, so both of the fills have the same tx and the same totalFilledAmount.

I would probably check: `depositId`, `originChainId`, AND `totalFilledAmount`

Reply:
> Imagine one transaction hash has multiple fills for different deposits

Wouldn't it emit different events for each deposit?

The `FillEventsConsumer` performs this check of each deposit individually

Reply:
Yes, you're right each deposit is a unique event. So i think you're right: for a given deposit the `totalFilledAmount` is unique!